third-culture-software/bhima
# Delete delay, initial_quantity, quantity, entry_date fields from the lot table

This PR removes fields from the lot table.
- The **delay** field was not used and was purged.
- The **initial_quantity** removed from the lot table.  It was ambiguous and not necessary and did not make sense to be part of the lot table.
- The **quantity** field was removed from the table.  Various queries were modified to compute the current quantity for the lot using the stock_movement table.
- and **entry_date** field was removed from the lot table.   Various queries were modified to get the date of the earliest stock movement for this lot as the lot's entry date.  Although we considered purging it entirely, it did seem useful when deciding whether to merge lots.

Closes https://github.com/IMA-WorldHealth/bhima/issues/5366

**Testing**
- Suggest using a production database.
- Try a few stock entries and exists
- Try editing a lot (from the lot registry).  Notice that there all these fields are no longer present.
- Look at the Lots registry, hover over the header cell of the "Entry Date" column and notice the expanded tooltip
- Try Stock > Find Duplicate Lots (you may need to enable permission for this page)
   - Notice that there is no 'entry_date' column.  Since each line represents several lots with (potentially) different entry_dates, it did not make sense to display it here.
   - On the action menu of one duplicate lots, use the "Merge duplicate lots" item.  Notice that the popup does have the entry date of each lot (actually the date of the first stock movement for this lot).  I decided to leave this in since it might help with decisions about merging lots.

NOTE: This PR did not remove 'unit_cost' from the lot table.  This should probably be done but requires a little more discussion.  I'll create a new issue for it.
